src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -64,11 +64,10 @@ private LocalDefinition EmitAddress(BoundExpression expression, AddressKind addr
                     break;
 
                 case BoundKind.ThisReference:
-                    Debug.Assert(expression.Type.IsValueType || IsAnyReadOnly(addressKind), "'this' is readonly in classes");
+                    Debug.Assert(expression.Type.IsValueType || expression.Type.IsExtension || IsAnyReadOnly(addressKind), "'this' is readonly in classes");
```

Comment:
>IsValueType [](http://example.com/codeflow?start=49&length=11)

It feels like we need to audit all use sites of this API to determine if they should be adjusted in a similar way (not necessary in this PR). Adjusting implementation of ```IsValueType``` to return true for extension types could be an alternative. Is there a good reason to not do that?

Reply:
I have such audit as a TODO. I'll add a PROTOTYPE comment as well.
From a symbol/API perspective we've decided that extensions are a separate kind of type, and thus are not value types